sale_order_warehouse_location/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2025 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("partner_shipping_id")
+    def _compute_warehouse_id(self):
+        ret_vals = super()._compute_warehouse_id()
+        for order in self.filtered(
+            lambda o: o.partner_shipping_id
+            and (o.state in ["draft", "sent"] or not o.ids)
+        ):
+            warehouses = self.env["stock.warehouse"].search(
```

Comment:
Please resolve the class instance outside the loop:

```python
warehouse_obj = self.env["stock.warehouse"]
    ....
    warehouses = warehouse_obj.search(...)
```

I would also suggest to find a way to fetch all the warehouses using a single query and use "filtered" afterwards. This will significantly improve the performance.

Reply:
I've done some changes. Could you have a look, please? Thank you.